sdk/src/transaction_context.rs

```diff
@@ -215,11 +227,21 @@ impl TransactionContext {
 
     /// Pops the current InstructionContext
     pub fn pop(&mut self) -> Result<(), InstructionError> {
-        if self.instruction_stack.is_empty() {
-            return Err(InstructionError::CallDepth);
-        }
+        let instruction_context = self.get_current_instruction_context()?;
```

Comment:
(heh, my review comment is helping already: https://github.com/solana-labs/solana/pull/25899/files#r918481393)

this `?` (and another one below) can potentially create mismatched push() and pop(). is this really safe? could you comment the reason if this is so or adjust the code?